openpype/hosts/max/plugins/publish/extract_tycache.py

```diff
@@ -0,0 +1,197 @@
+        filenames = self.get_files(instance, start, end)
+        additional_attributes = instance.data.get("tyc_attrs", {})
+
+        with maintained_selection():
+            job_args = None
+            has_tyc_spline = (
+                True
+                if instance.data["tycache_type"] == "tycachespline"
+                else False
+            )
+            if instance.data["tycache_type"] == "tycache":
+                job_args = self.export_particle(
+                    instance.data["members"],
+                    start, end, path,
+                    additional_attributes,
+                    tycache_spline_enabled=has_tyc_spline)
+            for job in job_args:
+                rt.Execute(job)
+        representations = instance.data.setdefault("representations", [])
+        representation = {
+            'name': 'tyc',
+            'ext': 'tyc',
+            'files': filenames if len(filenames) > 1 else filenames[0],
+            "stagingDir": stagingdir,
+        }
+        representations.append(representation)
+        self.log.info(f"Extracted instance '{instance.name}' to: {filenames}")
+
+        # Get the tyMesh filename for extraction
+        mesh_filename = "{}__tyMesh.tyc".format(instance.name)
+        mesh_repres = {
+            'name': 'tyMesh',
+            'ext': 'tyc',
+            'files': mesh_filename,
+            "stagingDir": stagingdir,
+            "outputName": '__tyMesh'
+        }
+        representations.append(mesh_repres)
+        self.log.info(
+            f"Extracted instance '{instance.name}' to: {mesh_filename}")
+
+    def get_files(self, instance, start_frame, end_frame):
+        """Get file names for tyFlow in tyCache format.
+
+        Set the filenames accordingly to the tyCache file
+        naming extension(.tyc) for the publishing purpose
+
+        Actual File Output from tyFlow in tyCache format:
+        <InstanceName>__tyPart_<frame>.tyc
+
+        e.g. tycacheMain__tyPart_00000.tyc
+
+        Args:
+            instance (str): instance.
+            start_frame (int): Start frame.
+            end_frame (int): End frame.
+
+        Returns:
+            filenames(list): list of filenames
+
+        """
+        filenames = []
+        # should we include frame 0 ?
+        for frame in range(int(start_frame), int(end_frame) + 1):
+            filename = "{}__tyPart_{:05}.tyc".format(instance.name, frame)
+            filenames.append(filename)
+        return filenames
+
+    def export_particle(self, members, start, end,
+                        filepath, additional_attributes,
+                        tycache_spline_enabled=False):
+        """Sets up all job arguments for attributes.
+
+        Those attributes are to be exported in MAX Script.
+
+        Args:
+            members (list): Member nodes of the instance.
+            start (int): Start frame.
+            end (int): End frame.
+            filepath (str): Output path of the TyCache file.
```

Comment:
this is missing documentation for the rest of the arguments -- `additional_attributes` and `tycache_spline_enabled`